Feat: Add support for lifecycle hook (#638)
convox/convox: docs/reference/primitives/app/service.md

| **image**       | string     |                     | An external Docker image to use for this Service (supercedes **build**)                                                                      |
| **internal**    | boolean    | false               | Set to **true** to make this Service only accessible inside the Rack                                                                         |
| **internalRouter** | boolean    | false               | Set it to **true** to make this Service only accessible using internal loadbalancer. You also have to set the rack parameter [internal_router](/installation/production-rack/aws) to **true**                 |

> | **lifecycle** |  map  |       | Set container prestop and poststart hook. This gives you ability to run command before terminating and after starting the container |

Reviewer: could you elaborate a little bit more on the description?

```suggestion
| **lifecycle** |  map  |       | The prestop and poststart hooks enable running commands before terminating and after starting the container, respectively |
```

&nbsp;

### lifecycle

| Attribute | Type   | Default | Description                                                                                |
| --------- | ------ | ------- | ------------------------------------------------------------------------------------------ |
| **perStop**     | string |         | The pre stop command |
| **postStart**  | string |         | The post stop command |

> &nbsp;

Reviewer: is this intentional?
